management/dns_update.py

```diff
@@ -144,8 +144,12 @@ def build_zone(domain, all_domains, additional_records, www_redirect_domains, en
 		records.append((None,  "NS",  "ns1.%s." % env["PRIMARY_HOSTNAME"], False))
 
 		# Define ns2.PRIMARY_HOSTNAME or whatever the user overrides.
-		secondary_ns = get_secondary_dns(additional_records) or ("ns2." + env["PRIMARY_HOSTNAME"])
-		records.append((None,  "NS", secondary_ns+'.', False))
+		# User may provide one or more additional nameservers
+		if get_secondary_dns(additional_records):
+				[records.append((None,  "NS", secondary_ns+'.', False)) for secondary_ns in get_secondary_dns(additional_records)]
```

Comment:
Whoa. You put a call inside a generator. No, that's too award. Please replace with a regular for loop.